test/functional/example_test.py

```diff
@@ -80,11 +80,11 @@ def set_test_params(self):
         # Set setup_clean_chain to True to skip this and start from the Genesis
         # block.
         self.setup_clean_chain = True
-        self.num_nodes = 3
-        # Use self.extra_args to change command-line arguments for the nodes
-        self.extra_args = [[], [], []]
-
-        # self.log.info("I've finished set_test_params")  # Oops! Can't run self.log before run_test()
+        self.num_nodes = 2
+        self.extra_args = [
+            ["--p2p-addr=/ip6/::1/tcp/3031"],
```

Comment:
OK. Now I see the issue with ports on another level. These parameters should be passed in the initial configuration of launching a node. Not in individual tests, through the `p2p_port` variable.

So the work flow should be:

1. You launch the node, giving it the parameter `p2p_port` + the node number
2. Now the node obeys and uses the given port number
3. The `connect_node` function uses `p2p_port` to reproduce the same port number for the same node number
4. In the test, you just say specify the connection topology, like `connect_nodes(0,1)`, and the rest is done automatically through the internal configuration.